pkg/apply/solver/solver.go

```diff
@@ -232,7 +239,10 @@ func (t *TaskQueueBuilder) AppendPruneWaitTasks(pruneObjs []*unstructured.Unstru
 	if o.Prune {
 		// Use the "depends-on" annotation to create a graph, ands sort the
 		// objects to prune into sets using a (reverse) topological sort.
-		pruneSets := graph.ReverseSortObjs(pruneObjs)
+		pruneSets, err := graph.ReverseSortObjs(pruneObjs)
+		if err != nil {
+			t.err = err
```

Comment:
This feels a bit awkward and probably suggests that we are using the builder pattern in an unusual way here. Normally you would expect that the set/configure functions doesn't do any work, but just configures the builder and then all the actual work will be done in the `Build` function. 
With the current approach, it seems like we might want to check the `err` property on the builder on every call, so we can just exit immediately.

Reply:
I agree it is a bit awkward. But I'd prefer to update the `Builder` pattern in another PR if that's our path. Otherwise, this PR will be a lot larger. I'll add a TODO.